### PR TITLE
Add `TestTask::getTestDescriptor` method

### LINKED ISSUE #4711: Add `TestTask::getTestDescriptor` method
For our integration tests I needed a different parallelization algorithm than the one implemented on top of forkjoin which is shipped with the jupiter engine. While it was quite easy to implement this with the given APIs (besides the parallelization, it just used 99% of the jupiter-engine for everything else), I have found it a bit tricky to find out which TestNode is represented by each TestTask (could have used reflection, but used another trick).

I know this a very uncommon scenario, but would anything speak against simply adding a getTestDescriptor method to the TestTask to make usecases like this a bit easier?
I don't think there is anything wrong with this, as the TestTask always represents a TestNode which basically will always have a matching Descriptor, at least when I look at the default implementation shipped with junit. To avoid breaking changes it could be a default implementation which just throws an UnsupportedOperationException if not implemented.

Thanks in advance and keep up the good work, JUnit5 is really a great testing framework!

## Deliverables

- [x] Add `TestTask.getTestDescriptor()` method

### PR DESCRIPTION
**Pull request by eric6iese**
## Overview

Added TestTask::getTestDescriptor as described in the issue. A TestTask should always correspond to an specific test node, so making its descriptor public for further analysis in custom test engines could be useful.

Resolves #4711.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://docs.junit.org/snapshot/user-guide/) and [Release Notes](https://docs.junit.org/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/HierarchicalTestExecutorService.java [hunk 1]
```diff
@@ -95,6 +97,16 @@ interface TestTask {
 		 */
 		ResourceLock getResourceLock();
 
+		/**
+		 * Get the {@linkplain TestDescriptor test descriptor} of this task.
+		 *
+		 * @throws UnsupportedOperationException if not supported for this TestTask implementation
+		 */
```
Comment (marcphilipp) [inline junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/HierarchicalTestExecutorService.java:105]:
```suggestion
		 * @since 6.0
		 */
```

**Review by marcphilipp — APPROVED** [review]

**Comment by marcphilipp** [comment]
I tried editing your PR (by pushing to the branch) to add this to the release notes but GitHub wouldn't let me. I think that's because you opened the PR from the default branch of your fork.

I've added this to the release notes in a separate commit now: 5b3448d67b9f397e478098286fd56faca9a86e63
